### ISSUE TITLE
could not load type "ckeditor" (500 internal server error)

### ISSUE THREAD

**Issue by miltone** [egeloen/IvoryCKEditorBundle#98]
Hello,

I want to implement the ckeditor for symfony2 and I follow the installation instructions as 
1/ require in composer
2/ update composer
3/ register the bundle in AppKernel.php

my builder is this :

```
        $builder->add('texte','ckeditor', array(
            'config' => array(
                'toolbar' => array(
                    array(
                        'name'  => 'document',
                        'items' => array('Source', '-', 'Save', 'NewPage', 'DocProps', 'Preview', 'Print', '-', 'Templates'),
                    ),
                    '/',
                    array(
                        'name'  => 'basicstyles',
                        'items' => array('Bold', 'Italic', 'Underline', 'Strike', 'Subscript', 'Superscript', '-', 'RemoveFormat'),
                    ),
                ),
                'uiColor' => '#ffffff',
            )
        ));
```

but the result webpage is a symfony error that explain : could not load type "ckeditor"

```
CRITICAL - Uncaught PHP Exception Symfony\Component\Form\Exception\InvalidArgumentException: "Could not load type "ckeditor"" at C:\dev\Apache2.4.6\htdocs\MyBundle\vendor\symfony\symfony\src\Symfony\Component\Form\FormRegistry.php line 89 
```

Do you know what can do this ?

Thanks for your support

**Comment by ephrin** [comment]
Same for me. Does not work with string type definition. Tags was not fetched (registered). 
Cleared cache, redumped container. Only after direct form builder dependency injection of "ivory_ck_editor.form.type" and using it as an argument of field add as type it starts to work. (sy v2.7.7)

**Comment by egeloen** [comment]
Please if you have an issue don't comment on a closed one as I have poor chance I see it... Anyway, you're issue should be fixed by #199 and is a duplicate of #197

**Comment by ephrin** [comment]
:+1:

**Comment by napestershine** [comment]
@ephrin How did you fixed the error can you explain please?

@egeloen  Can you tell me how to fix the error, I am new in Symfony Thanks

**Comment by ephrin** [comment]
@napestershine you can define explicit dependency of ` "@ivory_ck_editor.form.type"`, for example, as a constructor argument of your form type. And then use it in your `FormInterface->add()` method as type. You can always use an instance of type instead of its alias when building a form.

**Comment by egeloen** [comment]
There is maybe an issue in the BC layer for a very specific Symfony version since I was relying on methods added in Symfony itself for making such BC layer.

So, Please reopen an issue, don't comment on closed one... And give as uch informations you can such as Symfony version, PHP version, your form type and the exact error message (with a stacktrace is even better).

**Comment by napestershine** [comment]
@egeloen  Thanks for looking into the issue. I have solved the problem by adding formtype in formtype.php.

**Comment by cesarluna** [comment]
Saludos,
Le comparto mi experiencia.
`Dentro de la clase: App\ExampleBundle\Form\MyFormType`
agregue:
`use Ivory\CKEditorBundle\Form\Type\CKEditorType;`

y al llamar al type:
`$builder`
 `           ->add('reference',CKEditorType::class, array(`
`    'filebrowsers' => array(`
`        'VideoBrowse',`
`    ),`
`))`
Aun no entiendo porque no carga el  type "ckeditor" estoy trabajando con symfony 3.2

**Comment by egeloen** [comment]
@cesarluna Have you enabled the bundle in your AppKernel? This is the first reason you can get the type not loaded.

**Comment by cesarluna** [comment]
Yes, is enable the bundle in my AppKernel.

`new Ivory\CKEditorBundle\IvoryCKEditorBundle(),`

**Comment by noherczeg** [comment]
The solution for me was to replace the string typed type declaration to the class based one:

From:
```
$builder
            ->add('fieldName', 'ckeditor', ...
```

To:
```
use Ivory\CKEditorBundle\Form\Type\CKEditorType;

...

$builder
            ->add('fieldName', CKEditorType::class, ...
```
